File: src/main/java/com/sothawo/springdataelastictest/foo/FooRouter.java
/*
 * (c) Copyright 2023 sothawo
 */
package com.sothawo.springdataelastictest.foo;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.function.server.RouterFunction;
import org.springframework.web.reactive.function.server.ServerResponse;

import static org.springframework.web.reactive.function.server.RequestPredicates.*;
import static org.springframework.web.reactive.function.server.RouterFunctions.*;

/**
 * @author P.J. Meisch (pj.meisch@sothawo.com)
 */
@Configuration("FooRouterConfiguration")
public class FooRouter {

	@Bean
	public RouterFunction<ServerResponse> fooRouter(FooHandler handler) {
		return nest(path("/foo"),
			route(GET("/test"), handler::test)
			);
	}
}

File: src/main/kotlin/com/sothawo/springdataelastictest/so/FooRouter.kt
package com.sothawo.springdataelastictest.so

import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.function.server.coRouter

/**
 * @author P.J. Meisch (pj.meisch@sothawo.com)
 */
@Configuration("FooRouterConfiguration")
class FooRouter {

	fun fooRouter(handler: FooHandler) = coRouter {
		"/foo".nest {
			GET("/test", handler::test)
		}
	}
}
